### ISSUE TITLE
Tracing is not working for google stackdriver

### ISSUE THREAD

**Issue by cqx820** [open-telemetry/opentelemetry-rust-contrib#8]
Hi,

I'm trying to send trace data to Google Cloud Trace, I tried a python script with a credential file and it's working for me, but I'm unable to do the same with rust. The `Cargo.toml` looks like 
```
opentelemetry = { version = "0.16.0", default_features = false, features = ["rt-tokio", "rt-async-std", "rt-tokio-current-thread", "trace", "metrics"] }
opentelemetry-stackdriver = { version = "0.12.0", default_features = false, features = ["gcp_auth", "yup-authorizer"] }
tracing = "0.1.28"
tracing-futures = "0.2.5"
tracing-opentelemetry = "0.15.0"
tracing-subscriber = "0.2.24"
```

My workflow is like this: 
1. Get exporter
```
let handle = tokio::runtime::Handle::current();
let spawner = tokio_adapter::TokioSpawner::new(handle);
let exporter = StackDriverExporter::connect(
    GcpAuthorizer::new().await.unwrap(), &spawner, None, None).await.unwrap();  // Tried YupAuthorizer as well, didn't work too.
```
2. Get tracer 
```
let tracer = sdk::trace::TracerProvider::builder()
      .with_simple_exporter(exporter)
      .build()
      .tracer("tracer_name", Some(VERSION));
```
3. Get subscriber 
```
let subscriber = tracing_subscriber::registry().with(tracing_opentelemetry::layer().with_tracer(tracer));
```
4. Set global default
```
tracing::subscriber::set_global_default(subscriber);
```
5. Call `tracing::span!(tracing::Level::TRACE, span_name, ...)` 

Above is my entire workflow, after running I can't find any traces on Google Cloud Trace, I'm sure the cred file has enough permission. I've also tried raising the trace levels but no lucks. There were no runtime errors found. I wonder if I'm doing wrong with getting the exporter and the tracer. 

I also wrote some sample codes below just for testing, but they didn't work for me too 
```
let exporter = StackDriverExporter::connect(
    GcpAuthorizer::new().await.unwrap(),
    &spawner, Some(Duration::from_secs(5)), Some(5usize),
  ).await.unwrap();

let tracer = sdk::trace::TracerProvider::builder()
      .with_simple_exporter(exporter)
      .build()
      .tracer("tracer_name", Some(VERSION));

let attributes =
      vec![KeyValue::new("version", "0.0.0")];

let _span = tracer
        .span_builder("my_test_span")
        .with_attributes(attributes)
        .start(&tracer);

tracer.with_span(_span, |cx| {
      sleep(Duration::from_millis(500));
    });
```    
Any suggestions would be appreciated, thank you very much!!

**Comment by cyberbudy** [comment]
I have the same problem - no traces with stackdriver exporter

I've tried to found source code, but links show 404

**Comment by ivan-brko** [comment]
I got everything to work with the latest versions and documented it [here](https://www.dotlambda.io/post/exporting-logs-and-traces-to-gcp).
I also linked the small repo where I put it all together and it correctly exports logs in my GCP Cloud Run application (as you can see in the screenshots), so you can look there. 

Some things could probably be improved, but it should work.

**Comment by ivan-brko** [comment]
@djc thanks for all the work you did for `opentelemetry-stackdriver`, it was great to be able to export traces to GCP rather quickly. 

Do you know if there is somewhere I can read the overview of the current state and plans for the crate? 
GitHub thread or something?  
Currently, the [docs](https://docs.rs/opentelemetry-stackdriver/latest/opentelemetry_stackdriver/) are not up to date (for example, it's missing `GcpAuthorizer`), if you follow the link for the repo in the docs it leads you to `https://github.com/open-telemetry/opentelemetry-rust` which doesn't have the code for `stackdriver` (not in the main branch, anyway). 
I can also see that in this repo the Workspace `Cargo.toml`, `stackdriver` is commented out with the following comment: `TODO: Add back in once this relies on a published version`.

I'm willing to try to help if it's needed.

**Comment by djc** [comment]
@ivan-brko I'm using it at work so I'm also passively maintaining it. However, I'm no longer active as an opentelemetry-rust maintainer and (unrelated to that) this project got moved and apparently the opentelemetry-rust maintainers missed/postponed some things when moving this crate to its new home. PRs still welcome.

**Comment by ivan-brko** [comment]
@djc PRs should be made against this repo? 

Currently, I can't get `stackdriver` in this repo (`main` branch) to compile, don't know if I'm doing something wrong as I didn't investigate it, wanted to check if this code is even supposed to be built.
<img width="1253" alt="image" src="https://github.com/open-telemetry/opentelemetry-rust-contrib/assets/25389931/31536ef0-5d00-4191-aa90-9a59a3aa12ab">

I already made a [small PR](https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/22) to fix a documentation issue created here, would work on a couple of other improvements gladly (fix the compile issue I posted if needed, documentation improvements + things like [this issue](https://github.com/open-telemetry/opentelemetry-rust-contrib/issues/3)) as I would like to get my feet wet with Rust OSS anyway. 

If I understand correctly, `opentelemetry-stackdriver` in this repository wasn't used to publish the actual new versions of the crate yet?

**Comment by djc** [comment]
No -- unfortunately this repo was forked off the original repo only after some semver-incompatible changes to crates in the original repo were made. It might make sense to change the opentelemetry dependencies for opentelemetry-stackdriver to reference the upstream crates as a Git dependency instead for now so you can move forward here? I would probably pin them to the last commit before the extraction of the -contrib repo.

**Comment by ivan-brko** [comment]
Yeah, that sounds reasonable for the development of new features, but I actually might try fixing the issues with the current build. 
A new version of the crate cannot be deployed from this repo before fixing this problem anyway? Publishing a crate that depends on a specific commit would not work well with other libraries that depend on OTEL crates.

Do you maybe know if just fixing the compile errors would be enough or if there are other things to look into?

**Comment by ivan-brko** [comment]
@djc if you manage to catch some time, you can take a look at [the PR](https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/23), using the latest `opentelemetry` and `opentelemetry_sdk` versions and fixing compile issues.
